src/olympia/addons/models.py

```diff
@@ -1420,6 +1421,10 @@ def watch_status(old_attr=None, new_attr=None, instance=None,
     latest_version = instance.find_latest_version(
         channel=amo.RELEASE_CHANNEL_LISTED)
 
+    # Update the author's account profile visibility
+    if new_status != old_status:
+        [author.update_is_public() for author in instance.authors.all()]
```

Comment:
Not sure what you mean by reusing the `watch_addon_user` task - what task?

On a status update there is already a lot of things happening, including potentially files being moved and signed, so a couple of extra queries isn't going to be noticed.  Besides, there are vanishingly small numbers of add-ons with more than a handful (more than 1, really) of authors so we'd be handling an edge case that might not exist.